aziot/Cargo.toml

```diff
@@ -6,21 +6,39 @@ edition = "2018"
 
 
 [dependencies]
+anyhow = "1.0.34"
+async-trait = "0.1.42"
 backtrace = "0.3"
 base64 = "0.12"
+byte-unit = "4.0.9"
+chrono = { version = "0.4", features = ["serde"] }
+colored = "2.0.0"
 derive_more = "0.99.11"
+erased-serde = "0.3.12"
+futures = "0.3"
```

Comment:
`futures` is a re-export of `futures-core`, `futures-util`, and a bunch of other `futures-*` crates, and some features we don't care about enabled by default. You likely only need to depend on `futures-core` (for `Stream`) and `futures-util` (for `{Try,}{Future,Stream}Ext`), just like the other code in this repo. So I'm asking you to depend on `futures-core`, `futures-util` individually as required instead of the full `futures` crate.

Reply:
Ah, yes, sure. Will do.

Reply:
Actually, funny enough, it seems that I never actually use `futures` anywhere in my code. I've removed it entirely.

I'll also do a quick sweet through the other deps to see if there's anything else that can be culled.

Reply:
huh, looks like we've got quite a few stragglers scattered around the repo. 

Running `cargo udeps` shows the following:

```
unused dependencies:
`aziot-dps-client-async v0.1.0 (/home/danielprilik/Work/iot-identity-service/identity/aziot-dps-client-async)`
└─── dependencies
     ├─── "aziot-key-client"
     └─── "aziot-key-openssl-engine"
`aziot-hub-client-async v0.1.0 (/home/danielprilik/Work/iot-identity-service/identity/aziot-hub-client-async)`
└─── dependencies
     ├─── "aziot-key-client"
     └─── "aziot-key-openssl-engine"
`aziot-identity-common-http v0.1.0 (/home/danielprilik/Work/iot-identity-service/identity/aziot-identity-common-http)`
└─── dependencies
     └─── "aziot-key-common-http"
`http-common v0.1.0 (/home/danielprilik/Work/iot-identity-service/http-common)`
└─── dependencies
     └─── "lazy_static"
```

Can I trim these as part of this PR, or should we leave it 'till later?

Reply:
That's a nice tool. TIL

You can remove those now, but it might be nice to run `udeps` as part of `make test-release`, so if you want to do that you can remove these in that PR instead.

Reply:
The caveat is that `cargo udeps` require using a nightly Rust compiler, so it wouldn't be a trivial addition to the CI. We could totally add it, but idk if it's worth gunking up the pipelines even more by installing the nightly compiler alongside stable, and then also re-building the project with the nightly compiler just for `cargo udeps`.

In any case, I think i'll just cull these deps in this PR as well.